docs/snippets/CMakeLists.txt

```diff
@@ -18,8 +18,10 @@ endif()
 
 file(GLOB SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp"
                   "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp"
-                  "${CMAKE_CURRENT_SOURCE_DIR}/src/*.c")
-file(GLOB GPU_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/gpu/*.cpp")
+                  "${CMAKE_CURRENT_SOURCE_DIR}/src/*.c"
+                  "${DOCS_SOURCE_DIR}/articles_en/assets/snippets/*.cpp"
+                  "${DOCS_SOURCE_DIR}/articles_en/assets/snippets/*.c")
+file(GLOB GPU_SOURCES "${DOCS_SOURCE_DIR}/articles_en/assets/*.cpp")
```

Comment:
GPU sources does not seem to be correct.

In previous logic, sources contains generic files, while GPU sources contained only GPU specific. 
We need to preserve it.

Reply:
Moved GPU sources to separate folder, and adjusted the GPU_SOURCES line in the CMakeLists to keep the prevoius logic in https://github.com/openvinotoolkit/openvino/pull/25791/commits/4681e0536369bbf3cb7090a8ad8c9e129565a9e8